src/log.c

```diff
@@ -0,0 +1,92 @@
+#include <string.h>
+#include <stdarg.h>
+#include <syslog.h>
+#include <errno.h>
+
+#include "gophernicus.h"
+
+#if defined(LOG_UPTO)
+#define _LOG_UPTO LOG_UPTO
+#else
+#define _LOG_UPTO(pri)							\
+	(								\
+	((pri) == LOG_EMERG)   ? (LOG_MASK(LOG_EMERG)) :		\
+	((pri) == LOG_ALERT)   ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT)) : \
+	((pri) == LOG_CRIT)    ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT)) : \
+	((pri) == LOG_ERR)     ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR)) : \
+	((pri) == LOG_WARNING) ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING)) : \
+	((pri) == LOG_NOTICE)  ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE)) : \
+	((pri) == LOG_INFO)    ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE) | LOG_MASK(LOG_INFO)) : \
+	((pri) == LOG_DEBUG)   ? (LOG_MASK(LOG_EMERG) | LOG_MASK(LOG_ALERT) | LOG_MASK(LOG_CRIT) | LOG_MASK(LOG_ERR) | LOG_MASK(LOG_WARNING) | LOG_MASK(LOG_NOTICE) | LOG_MASK(LOG_INFO) | LOG_MASK(LOG_DEBUG)) : \
+	0)
+#endif
+
+static int _enable = 0;
+
+void log_init(int enable, int debug)
+{
+	if (!enable)
+		return;
```

Comment:
Also, something should be put in the readme to the effect that debug being enabled requires syslog being enabled.

Reply:
The current master assumes `syslog.h` exists, and logging `syslog` is enabled by default. I'm not sure `gophernicus -d -ns` is a common use case, but I can describe this behavior in the manpage if you wish.

Reply:
No, neither am I, but I would prefer to document something rather than leave it undocumented because it's an "uncommon use case".

Reply:
Fixed in  9c0b6a6.
```
     -d              Print debug output in syslog(3) and /server-status.
                     When -ns (disable syslog(3)) is used this option has no
                     effect.
```